src/Lib.P2P/Cryptography/PreSharedKey.cs

```diff
@@ -0,0 +1,167 @@
+            // Salsa20 function is not reversible thus increasing our security margin.
+            var encrypted = new byte[64];
+            var nonce = Encoding.ASCII.GetBytes("finprint");
+            var cipher = new Salsa20Engine();
+            cipher.Init(true, new ParametersWithIV(new KeyParameter(Value), nonce));
+            cipher.ProcessBytes(encrypted, 0, encrypted.Length, encrypted, 0);
+
+            // Then do Shake-128 hash to reduce its length.
+            // This way if for some reason Shake is broken and Salsa20 preimage is possible,
+            // attacker has only half of the bytes necessary to recreate psk.
+            return MultiHash
+               .GetHashAlgorithm("shake-128")
+               .ComputeHash(encrypted);
+        }
+
+        /// <summary>
+        ///   Generate a new value of the specified length.
+        /// </summary>
+        /// <param name="length">
+        ///   The length in bits of the new key value, defaults to 256.
+        /// </param>
+        /// <returns>
+        ///   <b>this</b> for a fluent design.
+        /// </returns>
+        public PreSharedKey Generate(int length = 256)
+        {
+            using (var rng = RandomNumberGenerator.Create())
+            {
+                Value = new byte[length / 8];
+                rng.GetBytes(Value);
+            }
+
+            return this;
+        }
+
+        /// <summary>
+        ///   Write the key to text stream.
+        /// </summary>
+        /// <param name="text">
+        ///   A text writer.
+        /// </param>
+        /// <param name="format">
+        ///   Determines how the key <see cref="Value"/> is formatted.  Can be
+        ///   "base16" or "base64".  Defaults to "base16".
+        /// </param>
+        /// <remarks>
+        ///   The key is writen as three lines 
+        ///   (1) the codec name "/key/swarm/psk/1.0.0/"
+        ///   (2) the base encoding  "/base16/" or "/base64/", 
+        ///   (3) the key value in the base encoding
+        /// </remarks>
+        public void Export(TextWriter text, string format = "base16")
+        {
+            text.WriteLine(codecName);
+            text.Write("/");
+            text.Write(format);
+            text.WriteLine("/");
+            switch (format)
+            {
+                case "base16":
+                    text.WriteLine(Value.ToHexString());
+                    break;
+                case "base64":
+                    text.WriteLine(Base64NoPad.ToBase64NoPad(Value));
+                    break;
+                default:
+                    throw new Exception($"Unknown encoding '{format}'.");
+            }
+        }
+
+        /// <summary>
+        ///   Read the key from the text stream.
+        /// </summary>
+        /// <param name="text">
+        ///   A text reader.
+        /// </param>
+        public void Import(TextReader text)
+        {
+            if (text.ReadLine() != codecName)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)